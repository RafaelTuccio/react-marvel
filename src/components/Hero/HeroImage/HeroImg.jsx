import { useEffect, useState } from "react";

const HeroImg = (props) => {
    let { img, extension, title } = props




    return(
        <div className="img-hero">
            
                <img src={img + extension} alt={title}/>
               
          
        </div>
    );

}

export default HeroImg;