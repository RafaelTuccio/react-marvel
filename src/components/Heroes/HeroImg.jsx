import { useEffect, useState } from "react";

const HeroImg = (props) => {
    let { img, extension } = props




    return(
        <div className="img-hero">
            
                <img src={img + extension} alt=""/>
               
          
        </div>
    );

}

export default HeroImg;