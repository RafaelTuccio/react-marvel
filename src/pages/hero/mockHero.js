

export const heroMock = 
            {
                id: 1009351,
                name: "Hulk",
                description: "Caught in a gamma bomb explosion while trying to save the life of a teenager, Dr. Bruce Banner was transformed into the incredibly powerful creature called the Hulk. An all too often misunderstood hero, the angrier the Hulk gets, the stronger the Hulk gets.",
                modified: "2020-07-21T10:35:15-0400",
                thumbnail: {
                    path: "http://i.annihil.us/u/prod/marvel/i/mg/5/a0/538615ca33ab0",
                    extension: "jpg"
                },
            }
        

        