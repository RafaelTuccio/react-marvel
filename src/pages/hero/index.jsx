import HeroHeader from "../../components/Hero/Header/HeroHeader";
import HeroData from "../../components/Cards/HeroData";
//import ComicMovie from "../../components/Cards/ComicMovie";
//import ComicCard from "../../components/Cards/ComicCard/ComicCard";
import Comics from "../../components/Comics/Comics";
import { useState } from "react";
import { getHeroByName } from "../../service/apiService";
import FormHero from "../../components/Hero/Form/FormHero";
import HeroImg from "../../components/Hero/HeroImage/HeroImg";
import ErrorComponent from "../../components/Error/ErrorComponent";

const Hero = () => {
    
    let [hero, setHero] = useState([])
    let [comics, setComcis] = useState([])
    let [img, setImg] = useState("")
    let [extension, setExtension] = (useState(""))
    let [error, setError] = useState(false)
    
    
    const heroHandle = (name) => {
        getHeroByName(name).then(result => {
            setComcis(result.comics)
            setHero(result.hero)
            setImg(result.hero.thumbnail.path);
            setExtension("."+ result.hero.thumbnail.extension)
        }
        
        ).catch(err => {
            console.error(err)
            setError(true)
        })
    }
    
    return(
        <div>
            <HeroHeader />
            <div>
                <FormHero heroHandle={heroHandle} />
                <div className="hero-container">
                    <div className="hero-main">
                        <HeroData hero={hero}/>
                        <HeroImg img={img} extension={extension} title={hero.name}/>
                    </div>
                    <span className="section-titulo">Últimos lançamentos</span>
                    <Comics comics={comics}/>
                </div>
            </div>
        </div>
    );
}

export default Hero;