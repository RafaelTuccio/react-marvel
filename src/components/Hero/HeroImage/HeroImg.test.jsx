import { render, screen } from '@testing-library/react';
import { heroMock } from '../../../pages/hero/mockHero';
import HeroImg from './HeroImg';

const hero = heroMock;

const img = hero.thumbnail.path;
const extension = hero.thumbnail.extension;

describe("<HeroImg />", () => {
    it("deve renderizar na tela", () => {
        const {debug} = render(<HeroImg img={img} extension={extension} title={hero.name}/>)
        debug()
    })

    it("deve dar match com o snapshot", () => {
        const { container } = render(<HeroImg img={img} extension={extension} title={hero.name}/>);
        expect(container.firstChild).toMatchSnapshot();
    })
})