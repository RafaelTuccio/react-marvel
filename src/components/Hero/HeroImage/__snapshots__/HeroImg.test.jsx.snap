// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroImg /> deve dar match com o snapshot 1`] = `
<div
  class="img-hero"
>
  <img
    alt="Hulk"
    src="http://i.annihil.us/u/prod/marvel/i/mg/5/a0/538615ca33ab0jpg"
  />
</div>
`;
